src/mtl_trainer.py

```diff
@@ -51,6 +56,54 @@ def __init__(
         self.data_args = data_args
         self.eval_datasets = eval_datasets
         self.test_datasets = test_datasets
+        self.eval_results = {}
+
+    def get_optimizers(
```

Comment:
What is the difference with huggingface implementation ?

Reply:
`num_warmup_steps` is calculated with `self.args.warmup_proportion * num_training_steps`. There is a comment in the code. The newest huggingface version has `warmup_ratio` but moving to a newer version will require a lot of refractoring.